<?xml version="1.0"?>
<launch>
  <!-- Load params-->
  <rosparam command="load" file="$(find puma_bringup_jetson)/config/jetson_params.yaml"/>
  
  <!-- Arduino -->
  <node name="arduino_node" pkg="rosserial_python" type="serial_node.py" output="screen" respawn="true" respawn_delay="2">
    <param name="port"                  value="/dev/ttyUSB0"/>
    <param name="baud"                  value="250000" />
  </node>
  <!-- Reverse manage -->
  <node 
    name="puma_reverse" 
    pkg="puma_reverse" 
    type="puma_reverse_node.py" 
    output="screen"
  />
  <!-- Parking manage -->
  <node 
    name="puma_parking" 
    pkg="puma_parking" 
    type="puma_parking_node.py" 
    output="screen"
  />
  <!-- Gps -->
  <node pkg="ublox_gps" type="ublox_gps" name="ublox" output="screen" clear_params="true" respawn="true" respawn_delay="5">
    <rosparam command="load" file="$(find puma_bringup_jetson)/config/zed_f9p.yaml"/>
    <remap from="ublox/fix"           to="puma/sensors/gps/fix"/>
    <remap from="ublox/fix_velocity"  to="puma/sensors/gps/fix_velocity"/>
  </node>
  <!-- IMU -->
  <include file="$(find puma_bringup_jetson)/launch/imu_bringup.launch"/>
  
  <!-- Monitor pc -->
  <include file="$(find puma_system_monitor)/launch/jetson_monitor.launch"/>
  <!-- Joy-->
  <include file="$(find puma_joy)/launch/puma_joy.launch"/>
  <!-- mode selector -->
  <node name="puma_control_mode" pkg="puma_control_mode_selector" type="control_mode_node.py" output="screen" respawn="true" respawn_delay="5"/>
  
  
</launch>